### PR TITLE
Implement typed plutus script example

### PR DESCRIPTION
**Pull request by Jimbo4350**
- Implement the ability to automagically execute a typed plutus script via `scripts/plutus/example-txin-locking-plutus-script.sh guessinggame`  
- Implement an end to end example of a typed plutus script

### CODE REVIEW

File: cardano-api-gen/src/Gen/Cardano/Api/Typed.hs [hunk 1]
```diff
@@ -46,21 +46,20 @@ module Gen.Cardano.Api.Typed
...
-                    StakePoolKey, PlutusScript(PlutusScriptSerialised),
-                    StakeCredential(StakeCredentialByKey),
-                    ProtocolParameters(ProtocolParameters),
-                    OperationalCertificateIssueCounter(OperationalCertificateIssueCounter) )
+import           Cardano.Api.Byron (KeyWitness (ByronKeyWitness), Lovelace (Lovelace),
+                   WitnessNetworkIdOrByronAddress (..))
+import           Cardano.Api.Shelley (Hash (ScriptDataHash), KESPeriod (KESPeriod),
+                   OperationalCertificateIssueCounter (OperationalCertificateIssueCounter),
+                   PlutusScript (PlutusScriptSerialised), ProtocolParameters (ProtocolParameters),
+                   StakeCredential (StakeCredentialByKey), StakePoolKey)
```
Comment (dcoutts) [inline cardano-api-gen/src/Gen/Cardano/Api/Typed.hs:54]: There are no "real" changes in this module. Can we avoid changing the file at all?

File: cardano-api-test/test/Test/Cardano/Api/Typed/Script.hs [hunk 2]
```diff
@@ -126,6 +128,13 @@ prop_roundtrip_ScriptData =
     sData <- H.forAll genScriptData
     sData === fromAlonzoData (toAlonzoData sData)
 
+prop_roundtrip_ScriptDataHash :: Property
+prop_roundtrip_ScriptDataHash =
+  H.property $ do
+    sData <- H.forAll genScriptData
+    hashScriptData sData === ScriptDataHash (Alonzo.hashData (toAlonzoData sData :: Alonzo.Data StandardAlonzo))
```
Comment (dcoutts) [inline cardano-api-test/test/Test/Cardano/Api/Typed/Script.hs:135]: What is the purpose of this test? The definition of `hashScriptData` is exactly what you have there: `ScriptDataHash . Alonzo.hashData . toAlonzoData`.

**Review by dcoutts — APPROVED** [review]
Looking good.

File: plutus-example/plutus-example/README.md [hunk 3]
```diff
@@ -17,16 +17,15 @@ The off chain code is used for transaction construction. In this case we constru
 
 ### Where can I learn about Plutus scripts in more detail?
 
-Our education director, Lars Brünjes, has an excellent series of [tutorials](https://youtu.be/IEn6jUo-0vU) on youtube. We will not attempt to provide an indepth explanation of Plutus in this repository.
+Our education director, Lars Brünjes, has an excellent series of [tutorials](https://youtu.be/IEn6jUo-0vU) on youtube. We will not attempt to provide an indepth explanation of [Plutus](https://docs.cardano.org/projects/plutus/en/latest/) in this repository.
 
 ## Tutorials
 
 We demonstate the following scripts:
```
Comment (newhoggy) [inline plutus-example/plutus-example/README.md:24]: `demonstate` ➡️  `demonstrate`

File: cardano-api-test/test/Test/Cardano/Api/Typed/Script.hs (hunk 2 above)
Comment (Jimbo4350) [reply cardano-api-test/test/Test/Cardano/Api/Typed/Script.hs:135]: I was debugging why the ledger was saying I was missing a datum hash. I can remove it.

**Comment by Jimbo4350** [comment]
bors merge

**Pull request merged**